src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -40,8 +42,9 @@ private void InitializeCancellationToken(TimeSpan timeout)
             }
             else if (timeout > TimeSpan.Zero)
             {
-                _cts = new CancellationTokenSource(timeout);
+                _cts = new CancellationTokenSource();
```

Comment:
Definitely agree.  If this isn't needed in most cases, it should definitely be opt-in and/or simply done at the call site where needed, since the implementation here isn't doing anything a caller couldn't do as well.